File: tasks/swift/easy/fibonacci.swift
// Swift - Easy

import Foundation

func fibonacci(n: Int) -> Int {
    if n <= 1 {
        return n
    }
    
    var a = 0
    var b = 1
    
    for _ in 2...n {
        let temp = a + b
        a = b
        b = temp
    }
    
    return b
}

// Test cases (Don't touch!)
let testCases: [Int] = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

for test in testCases {
    print("fibonacci(\(test)) = \(fibonacci(n: test))")
}

File: tasks/rust/easy/src/fibonacci.rs
// Rust - Easy

use std::io;

// TODO: Implement fibonacci sequence in Rust
fn main(){
    // Taking input of the number of terms from user.
    let mut term = String::new();
    io::stdin()
        .read_line(&mut term)
        .expect("Failed to read line");
    let mut term : u32 = term.trim().parse().expect("Please type a number!");
    println!("Fibonacci Sequence till {}th term:", term);

    // starting and next of sequence
    let mut starting : u32 = 0;
    let mut next :u32 = 1;

    loop {
        if term == 0 {
            break;
        }
        println!("{}", starting);
        let temp = starting + next;
        starting = next;
        next = temp;
        term -= 1;

    }
}